### PR TITLE
DPE-5642: breaking(build_charm.yaml): Pack on self hosted runners

### PR DESCRIPTION
**Pull request by dragomirp**
Switch build charm jobs to self-hosted runners

### CODE REVIEW

File: python/cli/data_platform_workflows_cli/craft_tools/collect_bases.py [hunk 1]
```diff
@@ -23,8 +23,8 @@
 
 logging.basicConfig(level=logging.INFO, stream=sys.stdout)
 RUNNERS = {
-    craft.Architecture.X64: "ubuntu-latest",
-    craft.Architecture.ARM64: "Ubuntu_ARM64_4C_16G_02",
+    craft.Architecture.X64: ["self-hosted", "linux", "X64", "large", "jammy"],
+    craft.Architecture.ARM64: ["self-hosted", "linux", "ARM64", "large", "jammy"],
```
Comment (dragomirp) [inline python/cli/data_platform_workflows_cli/craft_tools/collect_bases.py:27]: We can probably use other runner sizes.

File: .github/workflows/build_charm.yaml [hunk 2]
```diff
@@ -87,8 +87,11 @@ jobs:
     runs-on: ${{ matrix.base.runner }}
     timeout-minutes: 120
     steps:
-      - name: (GitHub-hosted ARM runner) Install libpq-dev
-        if: ${{ matrix.base.runner == 'Ubuntu_ARM64_4C_16G_02' }}
+      - name: Install pipx
+        run: |
+          sudo apt-get update
+          sudo apt-get install pipx -y
```
Comment (dragomirp) [inline .github/workflows/build_charm.yaml:93]: Pipx is missing from the IS runners and packages are not up to date.

File: python/cli/data_platform_workflows_cli/craft_tools/collect_bases.py (hunk 1 above)
Comment (dragomirp) [reply python/cli/data_platform_workflows_cli/craft_tools/collect_bases.py:27]: Since charmcraft will spin lxd containers, we can also try using noble.

**Comment by dragomirp** [comment]
Testen on https://github.com/canonical/postgresql-operator/pull/655 (ccc build) and https://github.com/canonical/pgbouncer-operator/pull/396 (clean build)

| Postgresql | main | self-hosted |
| -- | --------- | -- | 
| Jammy amd64 | 5m 45s | 7m 2s | 
| Jammy arm64 | 7m 28s | 9m 31s |

| Pgbouncer | main | self-hosted |
| -- | --------- | -- | 
| Focal amd64 | 11m 57s  | 16m 45s   | 
| Focal arm64 | 18m 56s | 22m 22s  |
| Jammy amd64 | 13m 26s  | 10m 51s  | 
| Jammy arm64 | 17m 53s  | 17m 18s  |

**Comment by taurus-forever** [comment]
@carlcsaposs-canonical I believe we are still on the same ground and should start moving to IS runners.
As a first step we can see the pgbouncer VM stability before merging this... what is our plan?

**Comment by dragomirp** [comment]
> > @carlcsaposs-canonical I believe we are still on the same ground and should start moving to IS runners. As a first step we can see the pgbouncer VM stability before merging this... what is our plan?
> 
> imo, we should only use this branch on pgbouncer VM as a temporary solution until charmcraftcache is refactored

Started https://github.com/canonical/pgbouncer-operator/pull/396 to switch PGB

**Comment by dragomirp** [comment]
Closing this since there's more to be discussed before the switch.

**Pull request closed without merging**